### PR TITLE
switch markdownlint container to markdownlint-cli2

### PR DESCRIPTION
**Pull request by tuminoid**
Also fix or locally ignore markdownlint issues.

- project-infra PR https://github.com/metal3-io/project-infra/pull/619 needs to merge for markdownlint here to pass

### CODE REVIEW

**Review by Sunnatillo — DISMISSED** [review]
/lgtm

**Review by smoshiur1237 — DISMISSED** [review]
/approve

**Comment by tuminoid** [comment]
/hold
Needs https://github.com/metal3-io/project-infra/pull/619 to merge first.

File: wow/github-workflow.md [hunk 1]
```diff
@@ -4,39 +4,48 @@
...
+Short (50 chars max or as subjected by open source repo practise) summary
 
-More detailed explainatory text, if necessary. Wrap it to about 72 characters or so.
-Write your commit message in the imperative: "Fix bug" and not "Fixed bug" or "Fixes bug".
+More detailed explainatory text, if necessary. Wrap it to about 72
+characters or so. Write your commit message in the imperative: "Fix bug"
+and not "Fixed bug" or "Fixes bug".
 
-Co-authored-by: My Name <my.name@est.tech>
+Signed-off-by: My Name <my.name@est.tech>
```
Comment (lentzi90) [inline wow/github-workflow.md:48]: nit: Show also how to add a co-author.
```suggestion
Signed-off-by: My Name <my.name@est.tech>
Co-authored-by: My Colleague <my.colleague@est.tech>
```

File: wow/github-workflow.md [hunk 2]
```diff
@@ -193,26 +211,30 @@ Instead the code is merged with the internal pull request.
...
 ```bash
 git log --oneline --no-merges ..main
 ```
 
-prints out all the commits in the `main` branch which aren't in the `release-0.3` branch in a below provided format:
+prints out all the commits in the `main` branch which aren't in the
+`release-0.3` branch in a below provided format:
 
-```
+```console
```
Comment (lentzi90) [inline wow/github-workflow.md:237]:
```suggestion
```text
```

**Review by lentzi90 — DISMISSED** [review]
Two nits, but I will put approve already
/approve

**Comment by tuminoid** [comment]
Nits fixed and bumped to newer release 0.9.0 from 2 days ago.

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Review by kashifest — APPROVED** [review]
/approve

**Comment by tuminoid** [comment]
/retest
Test new linters.

**Comment by tuminoid** [comment]
/override test-integration-metal3-dev-tools-centos  test-integration-metal3-dev-tools-ubuntu

**Comment by tuminoid** [comment]
Project-infra is missing markdownlint for this repo: adding it in https://github.com/metal3-io/project-infra/pull/623

**Comment by tuminoid** [comment]
/unhold
Good to go.

**Pull request merged**